src/solver/polynomialization.jl

```diff
@@ -0,0 +1,393 @@
+log(x)
+
+julia> trav_mult(unwrap(9*log(x)^2), x, Ref(false), log(x))
+log(x)
+
+# value of broken is changed here to true
+julia> trav_mult(unwrap(9*log(x+1)), x, Ref(false), log(x))
+log(x)
+```
+"""
+function trav_mult(arg, var, broken, sub)
+    args_arg = arguments(arg)
+    for (i, arg2) in enumerate(args_arg)
+        !iscall(arg2) && continue
+
+        oper = operation(arg2)
+        if oper === (^)
+            tp = trav_pow(args_arg, i, var, broken, sub)
+            sub = isequal(tp, false) ? sub : tp
+            continue
+        end
+
+        isequal(add_sub(sub, arg2, var, broken), false) && continue
+        sub = arg2
+    end
+    return sub
+end
+
+"""
+    add_sub(sub, arg, var, broken::Ref{Bool})
+
+Checks if arg contains var, and compares it directly to sub,
+if they're the same, returns true, if sub is 0 and arg contains
+a transcendental function in var, returns true. If both cases are
+not met, false is returned indicating that the previous function
+in the stack should not sub. ``add_sub`` also changes ``broken``
+to true if arg contains a transcendental function thats different from
+sub in any way.
+
+# Arguments
+- sub: The sub stored in memory in ``turn_to_poly``. 
+- arg: The arg to be compared to sub.
+- var: Num and should satisfy is_singleton
+- broken: A reference bool which the value of is altered if a different transc function is found than sub
+
+# Examples
+```jldoctest
+julia> add_sub(3^x, unwrap(3^x), x, Ref(false))
+true
+
+julia> add_sub(0, unwrap(log(x)), x, Ref(false))
+true
+
+# broken here changed to true
+julia> add_sub(log(x+1), unwrap(log(x)), x, Ref(false))
+false
+```
+"""
+function add_sub(sub, arg, var, broken::Ref{Bool})
+    if contains_transcendental(arg, var)
+        if isequal(sub, 0)
+            return true
+        elseif !isequal(sub, arg)
+            broken[] = true
+            return false
+        end
+    end
+
+    cond1 = n_occurrences(arg, var) > 0
+    cond2 = isequal(sub, arg)
+    return (cond1 && cond2)
+end
+
+"""
+    contains_transcendental(arg, var, n_occ=1)
+
+Simple helper function to check if arg contains a julia
+transcendental function and has number of occurrences of var = n_occ.
+
+# Arguments
+- arg: The arg to be checked. Should be of type ``SymbolicUtils.BasicSymbolic{Real}``.
+- var: Num and should satisfy is_singleton
+- n_occ: The number of occurrences that arg should contain. Default value is 1.
+
+# Examples
+```jldoctest
+julia> contains_transcendental(unwrap(log(x)), x)
+true
+
+julia> contains_transcendental(unwrap(x), x)false
+
+julia> contains_transcendental(unwrap(x^2), x)
+false
+
+julia> contains_transcendental(unwrap(1), x)
+false
+
+julia> contains_transcendental(unwrap(sin(x)), x)
+true
+```
+"""
+function contains_transcendental(arg, var, n_occ = 1)
+    !iscall(arg) && return false
+    arg_oper = operation(arg)
+
+    friends = [sin, log, log2, log10, cos, tan, asin, acos, atan, exp]
```

Comment:
Perhaps someone from this list is also a friend:

https://github.com/JuliaSymbolics/Symbolics.jl/blob/44146cec74fbd799fd67090132e15aed66ead421/src/linearity.jl#L15